src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelSchema.scala

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.dsde.firecloud.model
+
+import org.broadinstitute.dsde.firecloud.FireCloudConfig
+import spray.json._
+
+import scala.util.{Failure, Success, Try}
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: hussein
+ * Date: 07/24/2015
+ * Time: 14:18
+ */
+
+object ModelSchema {
+
+  def getTypeSchema(entityType: String): Try[EntityMetadata] = {
+    EntityTypes.types.get(entityType) match {
+      case Some(schema) => Success(schema)
+      case None => Failure(new RuntimeException("Unknown entity type: " + entityType))
+    }
+  }
+
+  def isCollectionType(entityType: String): Try[Boolean] = {
+    getTypeSchema(entityType).map(_.memberType.isDefined)
+  }
+
+  def getCollectionMemberType(collectionEntityType: String): Try[Option[String]] = {
+    getTypeSchema(collectionEntityType).map(_.memberType)
+  }
```

Comment:
It seems strange that I'm repeating myself here, but I guess that's what I get for writing a facade.